collectors/python.d.plugin/tomcat/tomcat.chart.py

```diff
@@ -95,6 +102,32 @@ def __init__(self, configuration=None, name=None):
         self.definitions = CHARTS
         self.url = self.configuration.get('url', 'http://127.0.0.1:8080/manager/status?XML=true')
         self.connector_name = self.configuration.get('connector_name', None)
+        self.parse = self.xml_parse
+
+    def xml_parse(self, data):
+        try:
+            return ET.fromstring(data)
+        except ET.ParseError:
+            self.debug('%s is not a valid XML page. Please add "?XML=true" to tomcat status page.' % self.url)
+            return None
+
+    def xml_single_quote_fix_parse(self, data):
+        data = single_quote_regex.sub(r"='\g<1>\g<2>'", data)
+        return self.xml_parse(data)
+
+    def check(self):
+        self._manager = self._build_manager()
+
+        raw_data = self._get_raw_data()
+        if not raw_data:
+            return False
+
+        if fix_tomcat_single_quote and single_quote_regex.search(raw_data):
```

Comment:
i dont get why we need `fix_tomcat_single_quote`, since it is always `True`
`single_quote_regex.search(raw_data)` should be enough

and change `return True` to `return bool(self.parse(raw_data))`

Reply:
It was a precaution to be able to disable the fix quickly in case it brought problems, maybe with a configuration option. But I can remove it if it's better.

Reply:
Yes, remove it

Reply:
Ok.

If I put `return self.parse(raw_data)` in `check` method, I still have `xml = self.parse(raw_data)` in the `_get_data` method ; is it a problem if parse is called twice ? Otherwise I could put the result in a `self.xml` variable, but it seems strange to have a content affectation in a check method.

Reply:
it is ok, `check` is called only once before start.

flow is
 - call `check()`, if check returns 'False` drop the job, otherwise run job in thread
 - every `update_every` call `job.get_data()`

Reply:
that is why i suggest to `return self.parse(raw_data)` - to make sure we can parse the data  before the start. And i suggest to use `bool` to explicitly show that function returns boolean and not the result of `self.parse` (techically bool is not needed, yes)